### PR TITLE
Custom wordlist

### PR DESCRIPTION
**Pull request by SecurityCze**
Hello,
this should close issue #1.

It provides the user option to add custom wordlists.

Currently saved configuration should be automatically converted to a new configuration. However due to limitation of how `XmlSerializer` handles unexpected nodes inside expected elements it is not the most performant solution. But it seems it will do the trick.
And with any new save of the configuration, it will be converted and no longer a problem.

It would be ideal of anyone could do a bit of beta testing before it is merged. It is entirely possible I completely missed something (C# is not my native tongue).

---

### Test it out:

You can build it identically to this repository.

If you do not want to compile yourself and you trust me. (But you probably shouldn't trust a random guy on the web) Here is a compiled version against Keepass 2.57: [KeePassDiceware_custom_wordlists.zip](https://github.com/user-attachments/files/16806154/KeePassDiceware_custom_wordlists.zip)

```
SHA256 checksums of my build:
49482b7b36f4bf58b8bec7e28fdf9210a6ec073dcfe4e53a9ac2c9a795ecf172  KeePassDiceware.dll
7003eee3dfb52065ed9d802ab257caaec7e420ab68406fa9ccc80fd870bf0dfc  KeePassDiceware.pdb
```
---

### Screenshots

Options form:
image

---
Word list form:
image

### CODE REVIEW

**Comment by nitz** [comment]
@SecurityCze — First I apologize for taking so long to get to this! I hope you've at the very least been using your branch in my delay.

Second, I want to express my extreme gratitude. The amount of effort you've put in here is awesome, and it's people like you who make working on open source things truly worth it.

I've given it some poking and prodding and it feels like everything is working exactly like it should. I'm gonna get this merged in and a new release posted. I imagine you're about to be the reason a lot of people are very happy!

**Comment by SecurityCze** [comment]
@nitz - I really appreciate these kind words, thank you. You have nothing to apologize for. We all realize you maintain this project in your spare time. If it takes few days, weeks, or months, so be it. Nothing bad is going to happen.

Some of the gratitude belongs to you too (Thank you!). Without your plugin, I couldn't even contribute. I must say that I extremely enjoyed working on it (let's do it again soon?).

Good to hear that everything was working for you. I downloaded the newest release, and based on first tests it seems to still work.

----

_Just as a sidenote._ Maybe we should update the README.md too.

- reflect the new looks
- mention that the wordlist is dynamically loaded (do not delete it after configuration)
- wordlist encoding - I have it tested with UTF8 (that works) and windows-1252 (doesn't work - blackboxes instead of local characters)

Not sure why I did not think of it before. Oh well...

----

I need to finally finish my dictionary. I can't seem to figure out the right combination of readability and size. It is either too large (hard to understand words) or too small (low entropy, so long passphrases). But at least it is now quite easy to test new lists.

And if I find some spare time (probably not in the near future), I would like to take a look at some more stuff. I am currently thinking about polishing up the dialogs (wordlists and saltsources do not behave 100% identically).

I was also thinking about the entropy calculations. I might have an idea how to estimate them quite nicely - but it only works when you generate them. I have no idea how to calculate them when viewing an existing entry.

----

**TLDR:** Thank you too! I couldn't contribute without your hard work. README might need some work. Hopefully we can cooperate again soon.
